test/extensions/clusters/aggregate/cluster_integration_test.cc

```diff
@@ -0,0 +1,304 @@
+#include "common/singleton/manager_impl.h"
+#include "common/upstream/cluster_factory_impl.h"
+#include "common/upstream/cluster_manager_impl.h"
+
+#include "extensions/clusters/aggregate/cluster.h"
+
+#include "test/common/upstream/utility.h"
+#include "test/mocks/protobuf/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/mocks/ssl/mocks.h"
+#include "test/test_common/environment.h"
+#include "test/test_common/simulated_time_system.h"
+
+using testing::AtLeast;
+using testing::DoAll;
+using testing::Eq;
+using testing::InSequence;
+using testing::Return;
+using testing::ReturnRef;
+using testing::SizeIs;
+
+namespace Envoy {
+namespace Extensions {
+namespace Clusters {
+namespace Aggregate {
+namespace {
+
+// The tests in this file are split between testing with real clusters and some with mock clusters.
```

Comment:
instead of copying this code from the cluster manager tests we should probably split the helper class out into its own file so it can reused

Reply:
Resolved.